ViAn/Project/Analysis/tag.cpp

```diff
@@ -1,112 +1,91 @@
 #include "tag.h"
 
 Tag::~Tag() {
-    for (auto it = m_intervals.begin(); it != m_intervals.end(); ++it){
-        delete *it;
+    for (auto it = tag_map.begin(); it != tag_map.end(); ++it) {
+        delete (*it).second;
     }
-    m_intervals.clear();
+    tag_map.clear();
 }
 
-/**
- * @brief Tag::add_frame
- * @param frame
- * Adds the newly tagged frame as a new poi
- * unless it's at the edge of a current poi or that frame already is tagged
- */
-void Tag::add_frame(int frame) {
-    add_interval(new AnalysisInterval(frame, frame));
+void Tag::add_frame(int frame, TagFrame* t_frame) {
+    m_unsaved_changes = true;
+    tag_map.insert(std::pair<int,TagFrame*>(frame, t_frame));
 }
 
-void Tag::remove_frame(int frame) {
-    remove_interval(new AnalysisInterval(frame, frame));
+bool Tag::find_frame(int frame) {
+    return tag_map.count(frame);
 }
 
-void Tag::add_interval(AnalysisInterval *an_interval){
-    BasicAnalysis::add_interval(an_interval);
-    merge_intervals();
+void Tag::remove_frame(int frame) {
+    auto it = tag_map.find(frame);
+    if (it != tag_map.end()) {
+        tag_map.erase(it);
+        delete (*it).second;
```

Comment:
Is this working ok? [Here](http://en.cppreference.com/w/cpp/container/map/erase) it says that the iterator will be invalidated after the erase

Reply:
I can look some more into this.